Nicer safe-select-keys error message if key not found

sbelak opened the pull request:
Made safe-select-keys return same error message as safe-get (ie. include the list of valid keys).

w01fe commented on src/plumbing/map.cljx:
```diff
@@ -14,7 +14,8 @@
   (doseq [k ks]
-    (assert (contains? m k)))
+    (or (contains? m k)
+        (schema/assert-iae false "Key %s not found in %s" k (mapv key m))))
```
`when-not` would be more idiomatic than `or`.  Or better yet, why not put the assertion in the assert-iae?

w01fe replied:
If there are more than one missing keys, it would also be nice to show them all.

w01fe replied:
Finally, I've been meaning to modify `safe-get` to ellipsize the keys, since the error message is not particularly nice when it fails on a map with millions of entries.  Not necessary to do that as part of this change, but bonus points if you're feeling up to it :).

sbelak replied:
Any preference as to when to ellipsize? I'm thinking 200.

w01fe commented:
Thanks for the PR!  Looks good in principle, just a minor comment / suggestion or two.

sbelak commented:
Changed to display all missing keys at once and truncate at 200 for large maps. 

The print-str is there to make the output look nice, otherwise string quotation marks gets printed making it look really hackish.

w01fe commented:
Looks/sounds great, thanks!

On Mon, Apr 4, 2016 at 12:00 AM, Simon Belak <notifications@github.com>
wrote:

> Changed to display all missing keys at once and truncate at 200 for large
> maps.
>
> The print-str is there to make the output look nice, otherwise string
> quotation marks gets printed making it look really hackish.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/plumatic/plumbing/pull/124#issuecomment-205024149>
>

sbelak commented:
Come to think of it, truncation can be solved much more elegantly using *print-length* (implemented in the latest patch).

Also added truncating safe-get and while I was at it, I changed merge-disjoint to return all conflicting keys on error.

w01fe commented:
All looks great, thanks again!
